back/app/services/user_service.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# This class exists to track and have in one place all operations
+# that write important user fields like email and confirmation.
+# It makes it easier to understand what and when can happen to user.
+#
+class UserService
+  class << self
+    def upsert_in_web_api(new_or_existing_user, user_params, &)
```

Comment:
All methods seem to accept a user_params argument. I therefore wonder if it could make sense to have a constructor that takes those params as its argument (perhaps together with an optional user), and turn the class methods into normal instance methods?

Reply:
Makes sense :) It's indeed not really a service to manage parameters.